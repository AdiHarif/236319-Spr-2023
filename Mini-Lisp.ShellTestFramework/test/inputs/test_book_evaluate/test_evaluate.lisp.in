(evaluate 'a '(a.T))
(evaluate 'a '(a.x))
(evaluate 'a '())
