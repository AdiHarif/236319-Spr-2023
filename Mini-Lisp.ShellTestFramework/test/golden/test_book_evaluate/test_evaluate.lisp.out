T
X
T
NIL
X
X
(X.Y)
NIL
(Y Z)
NIL
(X.Y)
(X.X)
T
NIL
T
Y
NIL
NIL
T
T
A
B
T
NIL
NIL
Traceback (most recent call last):
	From function EVALUATE:
		Args: ((ERROR NIL) ((T.T) (NIL)))
	From function COND:
		Args: (((ATOM S-EXPRESSION) (LOOKUP S-EXPRESSION A-LIST)) ((IS-ATOMIC (CAR S-EXPRESSION)) (EVALUATE-ATOMIC S-EXPRESSION A-LIST)) (T (APPLY (EVALUATE (CAR S-EXPRESSION) A-LIST) (CDR S-EXPRESSION) A-LIST)))
	From function EVALUATE-ATOMIC:
		Args: ((ERROR NIL) ((T.T) (NIL)))
	From function APPLY-ATOMIC:
		Args: (ERROR (NIL) ((T.T) (NIL)))
	From function COND:
		Args: (((EQ (QUOTE COND) ATOMIC) (EVALUATE-COND ACTUALS A-LIST)) (T (APPLY-EAGER-ATOMIC ATOMIC (EVALUATE-LIST ACTUALS A-LIST) A-LIST)))
	From function APPLY-EAGER-ATOMIC:
		Args: (ERROR (NIL) ((T.T) (NIL)))
	From function COND:
		Args: (((EQ ATOMIC (QUOTE ERROR)) (ERROR ACTUALS)) ((EQ ATOMIC (QUOTE EVAL)) (EVALUATE (CAR ACTUALS) A-LIST)) (T (APPLY-TRIVIAL-ATOMIC ATOMIC (CAR ACTUALS) (CAR (CDR ACTUALS)))))
	From function ERROR:
		Args: (NIL)
Error [(NIL)] in (ERROR ACTUALS)
Traceback (most recent call last):
	From function EVALUATE:
		Args: ((ERROR A) ((A.MY_ERR) (T.T) (NIL)))
	From function COND:
		Args: (((ATOM S-EXPRESSION) (LOOKUP S-EXPRESSION A-LIST)) ((IS-ATOMIC (CAR S-EXPRESSION)) (EVALUATE-ATOMIC S-EXPRESSION A-LIST)) (T (APPLY (EVALUATE (CAR S-EXPRESSION) A-LIST) (CDR S-EXPRESSION) A-LIST)))
	From function EVALUATE-ATOMIC:
		Args: ((ERROR A) ((A.MY_ERR) (T.T) (NIL)))
	From function APPLY-ATOMIC:
		Args: (ERROR (A) ((A.MY_ERR) (T.T) (NIL)))
	From function COND:
		Args: (((EQ (QUOTE COND) ATOMIC) (EVALUATE-COND ACTUALS A-LIST)) (T (APPLY-EAGER-ATOMIC ATOMIC (EVALUATE-LIST ACTUALS A-LIST) A-LIST)))
	From function APPLY-EAGER-ATOMIC:
		Args: (ERROR (MY_ERR) ((A.MY_ERR) (T.T) (NIL)))
	From function COND:
		Args: (((EQ ATOMIC (QUOTE ERROR)) (ERROR ACTUALS)) ((EQ ATOMIC (QUOTE EVAL)) (EVALUATE (CAR ACTUALS) A-LIST)) (T (APPLY-TRIVIAL-ATOMIC ATOMIC (CAR ACTUALS) (CAR (CDR ACTUALS)))))
	From function ERROR:
		Args: (MY_ERR)
Error [(MY_ERR)] in (ERROR ACTUALS)
